inc/dbmysql.class.php

```diff
@@ -154,10 +156,21 @@ function escape($string) {
     *
     * @throws GlpitestSQLError
     */
-   function query($query) {
+   function query($query, $new = false) {
       global $CFG_GLPI, $DEBUG_SQL, $SQL_TOTAL_REQUEST;
 
       $is_debug = isset($_SESSION['glpi_use_mode']) && ($_SESSION['glpi_use_mode'] == Session::DEBUG_MODE);
+
+      if ($new === false && ($is_debug || defined('TU_USER'))) {
```

Comment:
I do not like the idea of adding a parameter for this behaviour.

I think we should deprecate usage in this function and create a new **private** method, that consider the given query as secured, and call it from BDmysql methods that "compile" queries. Doing this will consider all raw `['INSERT', 'UPDATE', 'DELETE']` queries given from another class as deprecated, which seems to be what we want.

Reply:
Another solution is to use a code like this:
```
$caller = end(debug_backtrace(0, 2));
if ($caller && $callerClass = $caller['class'] && !in_array($callerClass, [DBmysql::class, DBmysqlIterator::class])) {
```
Problem is that DBmysqlIterator call this method, so we could not deprecate non provate usage.

Reply:
Change reverted.